src/transport/TcpTransport.cpp

```diff
@@ -213,15 +143,15 @@ void TcpTransport::timeoutcb(evutil_socket_t fd, short what, void *arg) {
 
 void TcpTransport::eventcb(struct bufferevent *bev, short what, void *ctx) {
```

Comment:
We need to consider this racing scenario: when this callback is executing, TcpTransport object has already been deleted. See TcpRemotingClient::CreateTransport

Reply:
yeah, some race in here, and i think it exist in early version. we can't convert shared_prt to pointer, but i will find other solution.

Reply:
The shared_ptr/weak_ptr solution I am proposing is like this:
1. TcpRmotingClient maintains a hash map between bufferevent* to weak_ptr of TcpTransport
2. bufferevent_setcb, pass TcpRemotingClient pointer in, aka, bufferevent_setcb(m_bufferEvent, readNextMessageIntCallback, NULL, eventcb, m_tcpRemotingClient);
3. In eventcb, lookup and test TcpTransport via weak_ptr::lock()

Reply:
I prefer to using smart pointer here as well.